science/exe.py

```diff
@@ -245,90 +267,335 @@ def parse(cls, value: str) -> AppInfo:
     value: str
 
 
-@_main.command()
-@click.argument("config", type=click.File("rb"), default="lift.toml")
+@dataclass(frozen=True)
+class LiftConfig:
+    file_mappings: tuple[FileMapping, ...] = ()
+    invert_lazy_ids: frozenset[str] = frozenset()
+    include_provenance: bool = False
+    app_info: tuple[AppInfo, ...] = ()
+    app_name: str | None = None
+
+
+pass_lift = click.make_pass_decorator(LiftConfig)
```

Comment:
This `click.make_pass_decorator` paired with `click.pass_context` below on the _lift group is a documented means to support structured chaining of sub-command configuration from the parent group(s) with sane isolation of parent group flags from sub-command knowledge.